File: lec4.cpp
#include<iostream>
#include<vector>
using namespace std;

void input(vector<int>& arr){
    for(int i=0;i<arr.size();i++){
        cin>>arr[i];
    }
}

void display(vector<int>& arr){
    for(int i=0;i<arr.size();i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}

 int span(vector<int>& arr){
        int myMax=arr[0];
        int myMin=arr[0];
        for(int i=1;i<arr.size();i++){
            if(arr[i]>myMax){
                myMax=arr[i];
            }
            if(arr[i]<myMin){
                myMin=arr[i];
            }
        }
        return myMax-myMin;
    }

    void printBar(vector<int>& arr){
        int myMax=arr[0];
        for(int i=1;i<arr.size();i++){
            if(myMax<arr[i]){
                myMax=arr[i];
            }
        }
        for(int i=myMax;i>0;i--){
            for(int j=0;j<arr.size();j++){
                if(arr[j]>=i){
                    cout<<"* ";
                }else{
                    cout<<"  ";
                }
            }
            cout<<endl;
        }
    }

    void swap(vector<int>& arr,int i,int j){
        int temp=arr[i];
        arr[i]=arr[j];
        arr[j]=temp;
    }

    void reverseArray(vector<int>& arr){
        int i=0;
        int j=arr.size()-1;

        while(i<j){
            swap(arr , i, j);
            i++; j--;
        }
    }

int main(){


    // int size=0;
    // cin>>size;

    vector<int> arr{3,5,7,1,4,2};
    // printBar(arr);
    display(arr);
    reverseArray(arr);
    // input(arr);
    display(arr);
    return 0;
}

File: lec4.java
import java.util.*;
class lec4{

    static Scanner scn=new Scanner(System.in);

    public static void input(int[] arr){
        for(int i=0;i<arr.length;i++){
            arr[i]=scn.nextInt();
        }
    }

    public static void display(int[] arr){
        for(int i=0;i<arr.length;i++){
            System.out.print(arr[i]+" ");
        }

    }

    public static int findMax(int[] arr){
        int myMax=arr[0];

        for(int i=1;i<arr.length;i++){
            if(myMax<arr[i]){
                myMax=arr[i];
            }
        }
        return myMax;
    }

    public static boolean isThere(int[] arr,int val){
        boolean exist=false;
        for(int i=0;i<arr.length;i++){
            if(arr[i]==val){
                exist=true;
                break;
            }
        }
        return exist;
    }

    public static void addOfTA(int[] a,int[] b){
        int mySize= a.length<b.length?b.length+1:a.length+1;

        int[] ans=new int[mySize];

        int i=a.length-1, j=b.length-1, k=ans.length-1;
        int carry=0;

        while(i>=0 && j>=0){
              int num=a[i]+b[j]+carry;
              carry=num/10;
              num=num%10;
              ans[k]=num;
            i--; j--; k--;
        }

        if(i>=0){
           while(i>=0){ 
            int num=a[i]+carry;
            carry=num/10;
            num=num%10;
            ans[k]=num;
            i--; k--;
           }
        }

        if(j>=0){
            while(j>=0){
                int num=b[j]+carry;
                carry=num/10;
                num=num%10;
                ans[k]=num;
                j--; k--;
            }
        }

        if(carry>0){
            ans[k]=carry;
        }

        display(ans);
    }




    public static void  main(String[] args){
        
        fn(5);
        
        // int size=scn.nextInt();
        // int [] arr=new int[size];
        // // int[] arr1={10,20,30,49};
        // input(arr);
        // // display(arr);
        // int val=scn.nextInt();
        // boolean exist = isThere(arr,val);
        // System.out.println(exist);
    }




}